### ISSUE TITLE
Color contrast in Angular Material Design

### ISSUE THREAD

**Issue by marcysutton** [angular/material#807]
Angular Material (and Material Design in general) has color contrast issues that must be addressed for users with low vision or difficulty with color perception. 

For example, white toolbar text on a blue background fails WCAG 2.0 AA. It has a ratio of 2.6:1, while 3:1 is required for Medium Bold and Large Non-Bold text: http://leaverou.github.io/contrast-ratio/#%2342a5f5-on-white Material Design's theme colors all have similar ratios.

<h2>Angular Material Color Contrast Audit</h2>
<h3><a href="http://www.w3.org/TR/WCAG20/#visual-audio-contrast">WCAG Level AA</a></h3>

**Global**
 * Toolbar: White text on blue (fails Medium Bold 3:1)

**Sidenav**
 * Current section title (fails Medium Bold 3:1)

**Bottom Sheet**
 * ‘Comment Actions’ label (fails Small 4.5:1)

**Buttons (fails Small 4.5:1, Medium 3:1)**
 * Primary text button on gray 
 * Disabled text button on gray
 * Warn text button on gray
 * Primary raised on gray (4.5:1 only)
 * Disabled raised on gray
 * Warn raised on gray (4.5:1 only)
 * Themed button on gray
 * Focused text button

**Checkbox  (fails Small 4.5:1, Medium 3:1)**
 * Green Checked icon
 * Empty icon
 * Checked disabled icon

**Radio Buttons**
 * Green icon (fails 3:1)

**Subheader**
 * Blue title on white (why is this blue anyway?) (fails 4.5:1)
 * Gray title on white (fails 4.5:1)

**Switch**
 * Disabled, active icon (fails 3:1)

**Textfield (fails AA Small and Medium Bold 3:1 and 4.5:1)**
 * label on white
 * label on dark
 * placeholder on white
 * placeholder on dark

**Tabs (fails Medium Bold 3:1)**
 * Tab: white on blue
 * Tab: disabled on blue (really bad)

**Toolbar (fails Medium 3:1)**
 * White text on green

Visual sidenav example:
![Material Design sidenav](https://cloud.githubusercontent.com/assets/1045233/5251945/76c91f3e-794e-11e4-9b17-f5ca1cc361b3.png)

Material buttons in [Color Contrast Analyzer](https://chrome.google.com/webstore/detail/color-contrast-analyzer/dagdlcijhfbmgkjokkjicnnfimlebcll?hl=en):
![Contrast Analyzer screenshot](https://cloud.githubusercontent.com/assets/1045233/5270828/9630fe98-7a23-11e4-9417-a183ae26f277.png)

**Comment by marcysutton** [comment]
@naomiblack should we close this issue since the internal Material Design team is on it?

**Comment by ThomasBurleson** [comment]
Fixed.
